decimal.go

```diff
@@ -81,21 +81,28 @@ func New(value int64, exp int32) Decimal {
 //
 func NewFromString(value string) (Decimal, error) {
 	var intString string
-	var exp int32
+	var exp int64
+
+	// Check if number is using scientific notation
+	eIndex := strings.IndexAny(value, "Ee")
+	if eIndex != -1 {
+		expInt, err := strconv.ParseInt(value[eIndex+1:len(value)], 10, 64)
```

Comment:
You don't need the `len(value)` in the slice bounds: http://play.golang.org/p/v4cw0g_TkK